main-support.ts

```diff
@@ -562,3 +666,104 @@ export function sendCurrentProgress(current: number, total: number, stage: Impor
     globals.winRef.setProgressBar(-1);
   }
 }
+
+const chokidar = require('chokidar');
+const async = require('async');
+
+const metadataQueue = async.queue(checkForMetadata, 8);
+
+let cachedFinalArray = [];
+
+function sendNewVideoMetadata(imageElement: ImageElement) {
+  imageElement = insertTemporaryFieldsSingle(imageElement);
+  globals.angularApp.sender.send(
+    'newVideoMeta',
+    imageElement
+  );
+  imageElement.index = cachedFinalArray.length;
+  cachedFinalArray.push(imageElement);
+  const screenshotOutputFolder: string = path.join(globals.selectedOutputFolder, 'vha-' + globals.hubName);
+
+  if (!hasAllThumbs(imageElement.hash, screenshotOutputFolder, globals.screenshotSettings.clipSnippets > 0 )) {
+    queueThumbExtraction(imageElement);
+  }
+}
+
+export function checkForMetadata(file, callback) {
+  console.log('checking metadata for %s', file.fullPath);
+  let found = false;
+  if (!deepScan) {
+    cachedFinalArray.forEach((element) => {
```

Comment:
No need to change, but FYI:

`.forEach` is awesome, but it can't be interrupted early. In case of a 10,000 array, if it finds a match in the first entry, it still goes through the rest. Because of that it would probably be useful to do a `for (...)` loop with a `break` 👍 

I'll adjust once the code merges 👌

Reply:
Yep, makes sense - not sure what I was thinking here! 😅